packages/api/src/external/commonwell/link/shared.ts

```diff
@@ -66,12 +113,26 @@ export async function autoUpgradeNetworkLinks(
 
   if (networkLinks._embedded && networkLinks._embedded.networkLink?.length) {
     const lola1Links = networkLinks._embedded.networkLink.flatMap(filterTruthy).filter(isLOLA1);
+    const lola2or3Links = networkLinks._embedded.networkLink
+      .flatMap(filterTruthy)
+      .filter(isLOLA2 || isLOLA3);
 
-    const requests: Promise<NetworkLink>[] = [];
+    const downgradeRequests: Promise<NetworkLink | undefined>[] = lola2or3Links.map(link =>
+      assessIfCWCQDuplicatesAndDowngrade(
+        link,
+        commonWell,
+        queryMeta,
+        commonwellPatientId,
+        commonwellPersonId,
+        executionContext
+      )
+    );
+    await Promise.allSettled(downgradeRequests);
 
+    const upgradeRequests: Promise<NetworkLink>[] = [];
     lola1Links.forEach(async link => {
       if (link._links?.upgrade?.href) {
-        requests.push(
+        upgradeRequests.push(
```

Comment:
We should also check if the `lola1Links` should be upgraded or not, right?

Reply:
I didn't change the upgrading logic at all. Just a variable name change

Reply:
I know. But are we not changing the logic to downgrade lola 2 if there's a duplicate? We should also not upgrade to lola 2 if there's a duplicate, right?

Reply:
right smart makes sense

Reply:
done